cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssFileController.java

```diff
@@ -0,0 +1,174 @@
+                if (listener != null) {
+                    listener.downloadStatusUpdated(initialDownloadData);
+                }
+                return initialDownloadData;
+            } catch (IOException e) {
+                throw new CompletionException(e);
+            }
+        });
+    }
+
+    Download executeInitialDownloadRequest(ApiConnectionInfo apiConnectionInfo, DownloadRequest downloadRequest, OAuth2Token token) throws IOException {
+        String jsonBody = CumulusObjectMapper.mapObjectToJson(downloadRequest);
+        HttpRequestExecutor executor =
+            new CumulusHttpRequestBuilderImpl(apiConnectionInfo, DOWNLOADS_ENDPOINT, token)
+                .enableHttp2()
+                .post()
+                .withBody(jsonBody)
+                .withMediaType(ACCEPT_HEADER_V1);
+        try (HttpRequestResponse response = executor.execute()) {
+            return CumulusObjectMapper.mapJsonToObject(response.getBody(), Download.class);
+        }
+    }
+
+    /**
+     * Download File to specified local file location.
+     *
+     * @param generatedDssFileDownloadData - Download object containing URL of file to download
+     * @param pathToLocalFile              - path in which file will be downloaded to
+     * @param listener                     - CumulusDssFileDownloadListener implementation that listens for download updates
+     * @return CompletableFuture           - Reference to async download process to be handled by caller
+     */
+    public CompletableFuture<Void> downloadToLocalFile(Download generatedDssFileDownloadData, Path pathToLocalFile,
+                                                       CumulusDssFileDownloadListener listener) throws CompletionException {
+        if (generatedDssFileDownloadData == null) {
+            throw new CompletionException(new IOException("Missing download Data"));
+        }
+        if (generatedDssFileDownloadData.getFile() == null) {
+            throw new CompletionException(new IOException("Missing DSS File URL for Download ID: " + generatedDssFileDownloadData.getId()));
+        }
+        return CompletableFuture.runAsync(() -> {
+            try {
+                CumulusFileDownloaderUtil.downloadFileToLocal(generatedDssFileDownloadData, pathToLocalFile, listener);
+            } catch (IOException e) {
+                throw new CompletionException(e);
+            }
+        });
+
+    }
+
+    private Download monitorAndRetrieveDownloadDataWhenComplete(ApiConnectionInfo apiConnectionInfo, Download initialDownloadData,
+                                                                CumulusDssFileGenerationListener listener) throws IOException {
+        DownloadsEndpointInput downloadsEndpointInput = new DownloadsEndpointInput(initialDownloadData.getId());
+        String progressIntervalVal = System.getProperty(PROGRESS_INTERVAL_PROPERTY_KEY);
```

Comment:
You can use `java.time` classes here. `Duration` for interval which would follow ISO-8601 and then your progress checks can be updated to use `Instant.now()` and check the Duration between. It makes the API more readable.

Reply:
Updated to use Duration